lib/material/material_ui.flow

```diff
@@ -389,124 +389,136 @@ SampleStates(
-
-  	// playing audio
- 	currIndexB = make(0);
-
-  	// init properties for each audio
- 	states = SampleStates(
- 		map(filenames, \__ -> make(0.0)),
- 		map(filenames, \__ -> make(0.0)),
- 		map(filenames, \__ -> make(false)),
- 		map(filenames, \__ -> make(false)),
- 	);
-
-  	synchrocalls = mapi(filenames, \i, __ ->
- 		SynchroCalls(
- 			// set ready
- 			\-> nextDistinct(states.isReady[i], true),
- 			// play next or stop in the end
- 			\-> {
- 				// reset position (for js)
- 				nextDistinct(states.position[i], 0.);
- 				// if last audio
- 				if (i == amount - 1) {
- 					if (!fgetValue(loopEx)) {
- 						// if not looping
- 						nextDistinct(playEx, false);
- 						synchrocallEx.onStop();
- 					}
- 				}
- 				// next for case with length == 1
- 				next(currIndexB, (i + 1) % amount);
- 			}
- 		)
- 	);
-
-  	players = mapi(filenames, \i, url ->
- 		TVideo(
- 			url,
- 			make(WidthHeight(-1., -1.)),		// not to draw
- 			replaceStructMany(style,
- 				[
- 					FVideoLength(states.length[i]),
- 					FVideoPosition(states.position[i]),
- 					FVideoPlay(states.isPlay[i]),
- 					synchrocalls[i],
- 				]
- 			)
- 		)
- 	);
-
-  	changeAudioPlaying = \index, isPlay -> {
- 		nextDistinct(states.isPlay[index], isPlay);
- 	}
-
-  	playerPos = fsubselect(
- 		currIndexB,
- 		FLift(\index ->
- 			faddition(
- 				fsum(subrange(states.length, 0, index)),	// sum of previous lengths
- 				states.position[index]					// current possition
- 			)
- 		)
- 	);
- 	playerPosChange = \pos -> nextDistinct(posEx, pos);
- 	userPosChange = \pos -> {
- 		allLength = fgetValue(fmerge(states.length));
- 		// last index ; length sum of previous audio
- 		lastEnd = foldi(allLength, Pair(-1, 0.), \i, acc, len -> {
- 			newEnd = acc.second + len;
- 			isNext = (acc.first + 1 == i);
- 			if (isNext && newEnd < pos) Pair(i, newEnd)
- 			else acc;
- 		});
-
-  		// reset current player
- 		currIndex = getValue(currIndexB);
- 		changeAudioPlaying(currIndex, false);
- 		nextDistinct(states.position[currIndex], 0.);
- 		// start new player
- 		newIndex = lastEnd.first + 1;
- 		nextDistinct(states.position[newIndex], pos - lastEnd.second);
- 		// next need when index has not changed (continue playing)
- 		next(currIndexB, newIndex);
- 	}
-
-  	MShow(
+ 	MShow(
  		const(i2b(amount)),
- 		// rendering (empty)
- 		MConstruct(
- 			[
- 				// when all audios ready
- 				makeSubscribe(fands(states.isReady, false), \isReady ->
- 					if (isReady) synchrocallEx.onStart()
- 				),
- 				// sum all audio lengths
- 				\-> fconnect(fsum(states.length), lenEx),
- 				// set possition
- 				\-> fBidirectionalLink(playerPos, posEx, playerPosChange, userPosChange),
- 				// not make2Subscribe due to looping in 1-element array
- 				\-> subscribe(currIndexB, \index ->
- 					changeAudioPlaying(index, getValue(playEx))
- 				),
- 				\-> subscribe(playEx, \isPlay ->
- 					changeAudioPlaying(getValue(currIndexB), isPlay)
- 				)
- 			],
- 			MGroup(players)
- 		)
+		{
```

Comment:
No point in moving everything inside MShow since this code will be executed anyway despite of any const(i2b(amount)) value.
Also MShow(const(...), ...) makes no sense, plain if (...) {...} else { TEmpty(); } will do the trick and won't mislead anyone but it's a pretension to the one who wrote it first not to you.

Reply:
Sure, thanks. Fixed.